cmd/iam.go

```diff
@@ -522,9 +522,12 @@ func (sys *IAMSys) ListPolicies(ctx context.Context, bucketName string) (map[str
 		return nil, errServerNotInitialized
 	}
 
-	<-sys.configLoaded
-
-	return sys.store.ListPolicies(ctx, bucketName)
+	select {
+	case <-sys.configLoaded:
+		return sys.store.ListPolicies(ctx, bucketName)
+	default:
+		return nil, errIAMNotInitialized
```

Comment:
This is a significant change in behavior. For systems that take time to load IAM it will give wrong behavior.

To cancel hanging requests you can listen for context cancellation, like...
```suggestion
	case <-ctx.Done():
		return nil, ctx.Err()
```

Reply:
Thank you for your reply.
I agree with you, but I also find a condition. If the Iam system fails to initialize, but the context is initialized successfully, the interface will still hanging. I suggest adding the judgment of the context on the basis of maintaining the default judgment.

Reply:
@polaris-megrez You are changing the behavior by adding the default case. Load time can be significant on some setups, so not blocking will break those, since it will now return errors for an undefined amount of time.

Reply:
Thank you for your advice.
You're right. I didn't consider such a design purpose. I'll change my pr.